### DISCUSSION TITLE
Storage Object Policies Error.

### DISCUSSION THREAD

**Discussion by iamindrazulfi** [supabase/supabase#8963] [Questions]
Hey Folks, I got this error when uploading to the storage.

```js
{statusCode: '42501', error: '', message: 'new row violates row-level security policy for table "objects"'}
```

Here is my code, it's always getting that error.

```js
const uploadAssets = async () => {
        const { data, error } = await supabase.storage.from("product-assets").upload("public/icon.png", icon)
        console.log(data, error)
    }
```

I know it's policies issue, but i have setup the policies.

<img width="863" alt="image" src="https://user-images.githubusercontent.com/78650493/190050757-75995876-1472-4691-bde1-df5cc7342b8a.png">

I have tried also this.

```js
create policy "Public Access Bucket Images Get"
on storage.objects for select
using ( bucket_id = 'product-assets' );

create policy "Public Access Bucket Images Post"
on storage.objects for insert
with check ( bucket_id = 'product-assets' );
```

But the issue still persist.
Would you guys help please ? :) 

Many Thanks!

**Comment by GaryAustin1** [comment]
Do you have any other policies on objects table that might be applying?
Edit:  Unlikely the issue as they would normally be used with an 'or' between them.

Edit2:
Do you have a bucket "product-assets" created?  You get a policy violation if the bucket does not exist.
